activerecord/lib/active_record/relation/finder_methods.rb

```diff
@@ -456,10 +454,16 @@ def find_some(ids)
       # 11 ids with limit 3, offset 9 should give 2 results.
       if offset_value && (ids.size - offset_value < expected_size)
         expected_size = ids.size - offset_value
+      else
+        ids = ids.first(expected_size) unless self.values[:order]
```

Comment:
@jmbejar you were completely right about the `.limit` problem, basically the `where` was returning records ordered by ID and we couldn't assume all of needed records will be there, so now this line aims to send to `where` all and only the needed IDs.
I also improved the `test_find_with_ids_and_limit` test using your idea and it seems to work better now.

Would love to have your review and opinion, thanks!